pkg/util/hlc/hlc.go

```diff
@@ -143,6 +331,37 @@ func (c *Clock) getPhysicalClockLocked() int64 {
 			c.mu.monotonicityErrorsCount++
 			log.Warningf(context.TODO(), "backward time jump detected (%f seconds)", float64(-interval)/1e9)
 		}
+
+		// Physical time should not cross the persisted future wall time (if futureWallTime is set)
+		if c.mu.futureWallTime != 0 && c.mu.lastPhysicalTime > c.mu.futureWallTime {
+			log.Fatalf(
+				context.TODO(),
+				"Cluster setting server.hlc.persist_future_wall_time does not allow physical time "+
+					"%d to be greater than persisted wall time %d.",
+				c.mu.lastPhysicalTime,
+				c.mu.futureWallTime,
+			)
+		}
+
+		toleratedJump := int64(ToleratedOffset(c.mu.maxClockJump))
+		if toleratedJump > 0 {
+			if interval > int64(toleratedJump/10) {
+				log.Fatalf(
+					context.TODO(),
+					"detected backward time jump of %f seconds is not allowed with tolerance of %f seconds",
```

Comment:
It will be useful if we decide to do a force NTP sync on server start-up. So a clock jump => immediate crash => force NTP sync. 
The crash is kept optional depending on whether we do this.

Reply:
I still don't understand how it could be useful because the in-process clock is anyway monotonic. If you can describe a concrete example of usefulness, it would help me.

Again, any assumptions about how NTP works seems to need more experimental evidence.

Reply:
There is no need to detect backward jumps if `persistFutureWallTime` feature is used. If that is not used, this could find use cases if the user fixes the wall time manually on CRDB startup. I'm ok removing the fatal for backward jumps since we will be using `persistFutureWallTime`